### ISSUE TITLE
Precompiled artifact not found

### ISSUE THREAD

**Issue by thejohncotton** [haubie/midiex#8]
Hey David! Excited to do some more testing (hoping to see if I can get this working with Nerves)
I am getting the following issue when I try to use `midiex` as a dependency:
```
Generated rustler_precompiled app
==> midiex
Compiling 10 files (.ex)

== Compilation error in file lib/midiex/backend.ex ==
** (RuntimeError) Error while downloading precompiled NIF: the precompiled NIF file does not exist in the checksum file. Please consider run: `mix rustler_precompiled.download Midiex.Backend --only-local` to generate the checksum file..

You can force the project to build from scratch with:

    config :rustler_precompiled, :force_build, midiex: true

In order to force the build, you also need to add Rustler as a dependency in your `mix.exs`:

    {:rustler, ">= 0.0.0", optional: true}

    lib/midiex/backend.ex:9: (module)
could not compile dependency :midiex, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile midiex --force", update it with "mix deps.update midiex" or clean it with "mix deps.clean midiex"
```

It gives me the same trouble when launching the livebook. Is there another repository that hosts the precompiled rust code?

![Screenshot 2024-09-10 at 3 36 23 PM](https://github.com/user-attachments/assets/a3e96cad-b966-430d-b25a-13b609d81130)

**Comment by haubie** [comment]
Hi @thejohncotton  thanks for letting me know. 

I just tried it with a fresh livebook and I get the same error too. It looks like I forgot to update the checksum file for the precompiled binaries before publishing on the v0.6.2 release. This should be fixed now in release v0.6.3, if you'd like to update your mix dependency to `{:midiex, "~> 0.6.3"}` and see how you go.

The precompiled packages are stored in github at: https://github.com/haubie/midiex/releases

Test run using v0.6.3 showing it's downloading correctly now (on Mac):
<img width="956" alt="image" src="https://github.com/user-attachments/assets/5f28a016-fcd1-46bd-871c-52ae573ff917">

I'm also keen to try it out with Nerves. I'd love to know how you go!  I had attempted to create a precompiled version for older 32bit RPI0 but had to put that aside for a while as it was taking a bit too much time to create a stable online precompiled build for it! I have a RPI5 which I'm intending to turn into a synth and hoping that will be a bit smoother.

**Comment by thejohncotton** [comment]
Awesome. I just tested with 0.6.3 in Livebook and it works great!
I'll let you know how the nerves stuff goes. I think the biggest issue (no surprise) is making sure that the proper linux dependencies are there (ALSA etc). The base firmware for rpi0-x doesn't include these so we might be looking at compiling a special base system for nerves audio compatible devices.
